### ISSUE TITLE
[Task 15] Megatron FP16 optimizer 配置去硬编码方案确认

### ISSUE THREAD

**Issue by huxy1225-cloud** [redai-infra/Relax#145]
我认领了任务 15，把我对问题的理解、实现范围和验证口径记录在这里。Draft PR 会同步提交，后续有反馈我再继续调整。

## 我对问题的理解

我看了 `relax/backends/megatron/model.py` 里的 `setup_model_and_optimizer()`。现在只要开启 FP16，下面四个值就会在创建 `OptimizerConfig` 时被重新写死：

```text
initial_loss_scale = 32768
min_loss_scale = 1
use_precision_aware_optimizer = True
store_param_remainders = False
```

所以即使用户从命令行传了其他值，最后也不会生效。前三项在当前 Megatron parser 中已经有对应参数，`store_param_remainders` 还没有从 Relax 的命令行暴露出来。

## 我准备这样改

- 我会复用 Megatron 现有的 `initial_loss_scale`、`min_loss_scale` 和 `use_precision_aware_optimizer` 参数，并补上 `--no-use-precision-aware-optimizer`。
- 我会增加 `--store-param-remainders` 和 `--no-store-param-remainders`，这样开启和关闭都能明确指定。
- 这四项在用户没有设置时先保留为 `None`，然后在 Megatron 校验前补上对应模式的默认值。
- FP16 缺参时继续沿用 Relax 现在的行为：`32768 / 1 / 开启 precision-aware / 关闭 param-remainders`，同时只输出一条 warning，把实际采用的参数列出来。
- BF16 等非 FP16 路径继续使用当前 Megatron 的默认值，不输出 FP16 fallback warning。
- 我会删除 `model.py` 中四项无条件覆盖，让显式传入的值原样进入 `OptimizerConfig`。
- 我会在 `scripts/training/text/run-qwen3-4B-fp16-8xgpu.sh` 中写明四项配置，并把脚本收到的额外参数放在命令末尾，方便用户覆盖脚本默认值。
- 中英文配置文档会一起更新。

对于 loss scale，我准备只做必要校验：两个值都必须是大于 0 的有限数值，并且 `min_loss_scale <= initial_loss_scale`。我不会额外限制为 2 的幂，因为当前 Megatron 接受任意正浮点数。

## 我准备怎么验证

```bash
python3 -m pytest -v tests/backends/megatron/test_fp16_optimizer_config.py
python3 -m pytest -q tests/backends/megatron
bash -n scripts/training/text/run-qwen3-4B-fp16-8xgpu.sh
pre-commit run --all-files
```

我会覆盖未配置、部分配置、全部显式配置、脚本参数覆盖、非法值、重复归一化不重复 warning、BF16 默认值，以及真实 `OptimizerConfig` 构造。

我的验证基线是 `cfda05351d4b137f60e2b71d048beab1316d9d9f`，使用官方镜像 `ghcr.io/redai-infra/relaxrl:dev-20260715-8325919e`，环境为 Python 3.12.3、PyTorch 2.11.0+cu129、CUDA 12.9.1 和 8 张 A100 80GB。

当前机器上没有这个脚本需要的 Qwen3-4B 模型、Megatron checkpoint 和 DAPO 数据，所以本次计划使用真实 Megatron parser、真实 `OptimizerConfig`、Megatron backend 回归和脚本检查作为验证证据。

## 想确认三点

1. 我把上面四项都放在任务 15 的修改范围内，这样理解是否正确？
2. 目前每个 arguments 实例最多输出一次缺参 warning，正常运行时通常对应每个进程一次。这个语义是否可以，还是要求只在分布式 rank 0 输出？
3. 在缺少完整训练资产的情况下，上面的 parser、`OptimizerConfig` 和 backend 回归是否足够作为验收证据？
